tests/models/faraday/test_faraday_model.py

```diff
@@ -141,3 +145,43 @@ def test_filter_mask():
     assert torch.equal(got_kwh, expected_tensor)
     for feature in got_features:
         assert torch.equal(got_features[feature], expected_tensor)
+
+
+def test_prepare_gmm_input():
+
+    vae_module = FaradayVAE(
+        class_dim=2, latent_dim=16, learning_rate=0.001, mse_weight=3
+    )
+    batch = TrainingData(
+        kwh=torch.rand(100, 48),
+        features={"feature_1": torch.rand(100), "feature_2": torch.rand(100)},
+        weights=torch.randint(low=1, high=5, size=(100,)),
+    )
+    encoded_batch = prepare_data_for_model(vae_module, batch)
+
+    assert encoded_batch.shape[0] == batch["weights"].sum()
```

Comment:
Would it be easy to check that the encoded batch actually has repeated rows in the way want it to, and not necessarily just got the right shape? 

So for the first row where weight !=1, check that this row is repeated `weight` times?

Reply:
Done, see latest commit. Note I had to abstract out the `prepare_data_for_model` function to allow for better testing.

Reply:
Nice, thank you for doing that 🙂